int32 id
string class_name
float32 speed